kratos/python_scripts/tikz_output_process.py

```diff
@@ -0,0 +1,246 @@
+        """ This method is executed before starting the time loop
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+
+        # First we retrieve the template argument
+        include_nodes_label = self.settings["include_nodes_label"].GetBool()
+        include_elements_label = self.settings["include_elements_label"].GetBool()
+        include_axis = self.settings["include_axis"].GetBool()
+        include_grid = self.settings["include_grid"].GetBool()
+        landscape_mode = self.settings["landscape_mode"].GetBool()
+        include_caption = self.settings["include_caption"].GetBool()
+        cmyk_colors_fill = self.settings["cmyk_colors_fill"].GetVector()
+        template_string = GetTemplateString(self.model_part, cmyk_colors_fill, include_nodes_label, include_elements_label, include_axis, include_grid, landscape_mode, include_caption)
+
+        # Generate nodal data
+        prefix_nodes = self.settings["prefix_nodes"].GetString()
+        nodal_data = ""
+        counter = 0
+        for node in self.model_part.Nodes:
+            if counter > 0:
+                nodal_data += "\n"
+            if prefix_nodes != "":
+                nodal_data += prefix_nodes + "_"
+            nodal_data += "{" + str(node.Id) + "}"
+            nodal_data += "    " + "{0:.12g}".format(node.X)
+            nodal_data += "    " + "{0:.12g}".format(node.Y)
+            counter += 1
+
+        # Generate elemental data
+        prefix_elements = self.settings["prefix_elements"].GetString()
+        elemental_data = ""
+        counter = 0
+        for elem in self.model_part.Elements:
+            if counter > 0:
+                elemental_data += "\n"
+            if prefix_elements != "":
+                elemental_data += prefix_elements + "_"
+            elemental_data += "{" + str(elem.Id) + "}"
+            for node in elem.GetNodes():
+                if prefix_nodes != "":
+                    elemental_data += prefix_nodes + "_"
+                elemental_data += "    {" + str(node.Id) + "}"
+            counter += 1
+
+        outputstring = template_string.replace("%%REPLACE_NODE_DATA", nodal_data)
+        outputstring = outputstring.replace("%%REPLACE_ELEMENT_DATA", elemental_data)
+        folder_name = self.settings["folder_name"].GetString()
+        model_part_name = self.settings["model_part_name"].GetString()
+        name_file = os.path.join(folder_name, model_part_name)
+        if self.output_control == "time":
+            time = self.model_part.ProcessInfo[KratosMultiphysics.TIME]
+            with open(name_file + "_TIME_" + str(time) + ".tex", 'w') as output:
+                output.write(outputstring)
+                output.close()
+        else:
+            step = self.model_part.ProcessInfo[KratosMultiphysics.STEP]
+            with open(name_file + "_STEP_" + str(step) + ".tex", 'w') as output:
+                output.write(outputstring)
+                output.close()
```

Comment:
```suggestion

```
not needed

Reply:
Is it because the with open?

Reply:
yes, this is the beauty of the with open
This way there won't be leaks if the execution crashes while writing or opening since it closes automatically
https://www.google.com/search?client=ubuntu&channel=fs&q=python+difference+with+open&ie=utf-8&oe=utf-8
There are some answers